app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/StatusReasonService.cs

```diff
@@ -0,0 +1,88 @@
+﻿using DynamicsAdapter.Web.SearchRequest.Models;
+using Simple.OData.Client;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using DynamicsAdapter.Web.Auth;
+using DynamicsAdapter.Web.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Quartz.Logging;
+
+namespace DynamicsAdapter.Web.SearchRequest
+{
+    public interface IStatusReasonService
+    {
+        Task<StatusReason> GetListAsync(CancellationToken cancellationToken);
+    }
+
+    public class StatusReasonService : IStatusReasonService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly OAuthOptions _oauthOptions;
+        private readonly ILogger<StatusReasonService> _logger;
+
+        public StatusReasonService(HttpClient httpClient, IOptions<OAuthOptions> oauthOptions,ILogger<StatusReasonService> logger)
+        { 
+            this._httpClient = httpClient;
+
+            _oauthOptions = oauthOptions.Value;
+            _logger = logger;
+        }
+
+        public async Task<StatusReason> GetListAsync(CancellationToken cancellationToken)
```

Comment:
OData interface does not work?

Reply:
yeah it doesnt work.. its mostly for entities and we are retrieving options sets on an entity